lizmap/modules/lizmap/classes/qgisForm.class.php

```diff
@@ -452,13 +452,188 @@ public function setFormDataFromFields($feature)
                 }
                 $form->setData($ref.'_hidden', $value);
             } else {
+                if (in_array(strtolower($this->formControls[$ref]->fieldEditType), array('date', 'time', 'datetime'))) {
+                    $edittype = $this->formControls[$ref]->getEditType();
+                    if (!$edittype || !property_exists($edittype, 'options')
+                        || !property_exists($edittype->options, 'field_format')) {
+                        break ;
+                    }
+                    $format = $this->convertQgisFormatToPHP($edittype->options->field_format);
+                    $date = DateTime::createFromFormat($format, $value);
+                    $value = $date->format('Y-m-d H:i:s');
+                }
```

Comment:
```suggestion
                if (in_array(strtolower($this->formControls[$ref]->fieldEditType), array('date', 'time', 'datetime'))) {
                    $edittype = $this->formControls[$ref]->getEditType();
                    if ($edittype && property_exists($edittype, 'options')
                        && property_exists($edittype->options, 'field_format')) {
                        $format = $this->convertQgisFormatToPHP($edittype->options->field_format);
                        $date = DateTime::createFromFormat($format, $value);
                        $value = $date->format('Y-m-d H:i:s');
                    }
                }
```
I was not sure that you can use a `break` here. @laurentj what do you think about it ?